src/dune_engine/scheduler.ml

```diff
@@ -597,7 +597,8 @@ end = struct
 
   let wait_unix t =
     Mutex.unlock t.mutex;
-    let proc_info = Proc.wait [] in
+    (* -1 means wait for any child process *)
```

Comment:
I dont think `wait_unix`s function signature should be changed to accept the `pid`, right? Not sure if the inline `(Pid.of_int (-1))` is fine here, am still relatively new to ocaml. let me know if that should be declared above and then re-used here or something